pkg/agent/controller/networkpolicy/cache.go

```diff
@@ -863,6 +870,11 @@ func (c *ruleCache) GetCompletedRule(ruleID string) (completedRule *CompletedRul
 
 	r := obj.(*rule)
 
+	if !c.antreaProxyEnabled && r.Direction == v1beta.DirectionOut && len(r.To.ToServices) != 0 {
+		klog.V(2).InfoS("Field `ToServices` can be only used when AntreaProxy is enabled", "ruleID", ruleID)
+		return nil, false, false
```

Comment:
We support reporting realization failure to NetworkPolicy condition. The error should be reported to antrea-controller to update the condition, instead of ignoring the rule without any error.

Reply:
Do we have any existing methods to report the message like this from agent to controller currently?

Reply:
We haven't done it before in agent but I think the method is there, just change `v1beta2.NetworkPolicyStatus` to reflect the failure. It could be generic for any failure we get from the reconciler.
BTW, I think `cache.go` is not a good place to perform the validation. It's implementation specific so should be validated by reconciler.go